cfme/containers/provider/openshift.py

```diff
@@ -109,16 +123,24 @@ def from_config(cls, prov_config, prov_key, appliance=None):
                 endpoints[endp] = OpenshiftDefaultEndpoint(**prov_config['endpoints'][endp])
             elif HawkularEndpoint.name == endp:
                 endpoints[endp] = HawkularEndpoint(**prov_config['endpoints'][endp])
+            # TODO Add Prometheus and logic for having to select or the other based on metrcis_type
             else:
                 raise Exception('Unsupported endpoint type "{}".'.format(endp))
 
+        if current_version() >= '5.9':
```

Comment:
**Required**
Use self.appliance.version here

Reply:
Though this requires accounting for when appliance isn't passed.

Reply:
Ignore this comment, we'll allow this for now.

Reply:
@jkrocil Suggested a better approach to this within your PR scope.

Don't try to version pick metrics_type here, just pass what's in the yaml (or None) via `prov_config.get()`.

Then in `OpenshiftProvider.__init__`, version pick based on the appliance version there.

Currently OpenshiftProvider has an optional appliance arg, but this will be a required option when converted to CEMv3.